caom2-artifact-resolvers/src/test/java/ca/nrc/cadc/caom2/artifact/resolvers/SubaruResolverTest.java

```diff
@@ -121,45 +123,55 @@ public void testGetSchema() {
 
     @Test
     public void testValidURI() {
-        try {
-            String uriStr = subaruResolver.getScheme() + ":" + RAW_DATA_URI + "/" + VALID_DATE1 + "/" + VALID_FILE1;
-            URI uri = new URI(uriStr);
-            URL url = subaruResolver.toURL(uri);
-            log.debug("toURL returned: " + url.toString());
-
-            String encodedValue = NetUtil.encode(VALID_DATE1 + "/" + VALID_FILE1);
-            Assert.assertEquals(url.toString(), PROTOCOL_STR + BASE_DATA_URL + DATA_URL_PATH + "?" + DATA_URL_QUERY +  encodedValue);
-            Assert.assertEquals(DATA_URL_PATH, url.getPath());
-            Assert.assertEquals(DATA_URL_QUERY + encodedValue, url.getQuery());
-            Assert.assertEquals(BASE_DATA_URL, url.getHost());
-
-            uriStr = subaruResolver.getScheme() + ":" + PREVIEW_URI + "/" + VALID_FILE2;
-            uri = new URI(uriStr);
-            url = subaruResolver.toURL(uri);
-            log.debug("toURL returned: " + url.toString());
-
-            Assert.assertEquals(url.toString(), PROTOCOL_STR + BASE_PREVIEW_URL + PREVIEW_URL_PATH + "?" + PREVIEW_URL_QUERY + VALID_FILE2);
-            Assert.assertEquals(PREVIEW_URL_PATH, url.getPath());
-            Assert.assertEquals(PREVIEW_URL_QUERY + VALID_FILE2, url.getQuery());
-            Assert.assertEquals(BASE_PREVIEW_URL, url.getHost());
-
-        } catch (Exception unexpected) {
-            log.error("unexpected exception", unexpected);
-            Assert.fail("unexpected exception: " + unexpected);
-        }
+        String uriStr = subaruResolver.getScheme() + ":" + RAW_DATA_URI + "/" + VALID_DATE1 + "/" + VALID_FILE1;
```

Comment:
Why did you remove the try/catch for unexpected exception? This is there so the stack trace gets printed and a dev can diagnose the failure. Otherwise, you just get something cryptic from junit... (same comment for other tests)

Reply:
These try/catch blocks hide the Exception that JUnit should throw itself and deliver to the reporter.  If an unexpected Exception happens, shouldn't it just be passed up?  Quite often when these tests are run and fail due to an unexpected Exception, only an ambiguous top level stack is printed, but the underlying one is not front and centre.